### PR TITLE
fix[venom]: fix branch eliminator cases in sccp

### PR DESCRIPTION
**Pull request by charles-cooper**
### Commit message

```
in sccp, when the operand of `jnz`, `djmp` or `assert` is already an
`IRLiteral` (this is most easily seen by disabling the IRnode branch
eliminator), the compiler will panic. this commit fixes the bug, and
refactor some commonly used code into helper functions.
```

### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

**Review by fubuloubu — APPROVED** [review]

File: vyper/venom/passes/sccp/sccp.py [hunk 1]
```diff
@@ -169,7 +170,8 @@ def _visit_expr(self, inst: IRInstruction):
             target = self.fn.get_basic_block(inst.operands[0].value)
             self.work_list.append(FlowWorkItem(inst.parent, target))
         elif opcode == "jnz":
-            lat = self.lattice[inst.operands[0]]
+            lat = _get_lattice(self.lattice, inst.operands[0])
```
Comment (harkal) [inline vyper/venom/passes/sccp/sccp.py:173]: I believe the `_get_lattice` name for the function is confusing, as what it actually does is either returns the actual lattice or if you are looking up a literal (which shouldn't happen) it returns that.

**Review by harkal — CHANGES_REQUESTED** [review]
`djmp` is also "vulnerable" to passing a literal. 

I think a better solution would be to never arrive here with literals in conditions. This will allow the algorithm to remain pure. A better place might be the `remove_unreachable_blocks` pass.

**Comment by charles-cooper** [comment]
how about literals inside of assertions?

File: vyper/venom/passes/sccp/sccp.py (hunk 1 above)
Comment (charles-cooper) [reply vyper/venom/passes/sccp/sccp.py:173]: there are other places in the file which guard lattice lookups with an `isinstance(..., IRVariable)` check

**Review by harkal — APPROVED** [review]

**Comment by charles-cooper** [comment]
i'm going to revert the changes to `exp` eval and the CI pipelines in order to keep this PR more "pure". will open those in a separate PR

**Pull request merged**